pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java

```diff
@@ -188,44 +188,32 @@ public NonPersistentTopic(String topic, BrokerService brokerService) {
 
     @Override
     public void publishMessage(ByteBuf data, PublishContext callback) {
```

Comment:
is this already running in the topic assigned executor?
It would be good to be able to assert that things are running in the ordered executor thread we expect them to.

Reply:
In this case, since this is non-persistent, we don't need to be in the ordered executor. We are in the IO thread from the connection of the producer and that's already enough to guarantee per-producer ordering.